### PR TITLE
Fix #26

### LINKED ISSUE #26: web-screening-app crashes on first time queuing
If a new user queues in for the first time, this will fail and he needs to try it a second time. This is the case for the production build of the React app, the development build will throw an error.

I guess the issue is somehow related to this: 
https://github.com/corona-school/backend-screening/blob/6d7099c0ae8b0f3e7eeeea64af3667dd8fd4be94/src/GenericQueue.ts#L95-L109

The for `forEach` should return, I guess. So probably someone should use a standard `for` loop or some fancy `reduce` or `flatMap` stuff.

### PR DESCRIPTION
**Pull request by geroembser**
This should fix #26 by replacing the misused `forEach` call in `getJobWithPosition`.

### CODE REVIEW

File: src/GenericQueue.ts [hunk 1]
```diff
@@ -94,18 +94,18 @@ export default class GenericQueue<D, S> extends EventEmitter {
 
   getJobWithPosition = async (id: string): Promise<JobInfo<D, S> | null> => {
     const currentList: Job<D, S>[] = await this.listInfo();
-    currentList
+    return currentList
       .filter((j) => j.status === "waiting")
       .sort((a, b) => a.timeWaiting - b.timeWaiting)
-      .forEach((job, index) => {
+      .reduce<JobInfo<D, S>>((acc, job, index) => {
```
Comment (Jonasdoubleyou) [inline src/GenericQueue.ts:100]: huh. What about `.find` :)

Comment (geroembser) [reply src/GenericQueue.ts:100]: A simple `find` would be even nicer. But I don’t know how to beautifully include the index in the result then. What’s your suggested way of implementing this? @LeonErath  doesn’t like this `reduce` approach either (which I thought was the best possible solution to keep this style), so I would appreciate another suggestion. Maybe the good old `for` loop, `map` or really something with `find`? 😂

Comment (Jonasdoubleyou) [reply src/GenericQueue.ts:100]:
```
const queue = ....sort(...);
const job = queue.find(it => it.id === id);
return { ...job, position: queue.indexOf(job) };
```

But it's kinda weird that the queue needs to be sorted :sweat_smile:

Comment (geroembser) [reply src/GenericQueue.ts:100]: Ok that's cleaner, true... But if there’s no job at all? Wouldn’t that lead to some problems? If not, then `getJobWithPosition` will never return `null`? Then we should also adjust the method's return type to be consistent.

Comment (Jonasdoubleyou) [reply src/GenericQueue.ts:100]: Well, the position is also kinda senseless then, so `if(!job) return null;` would work

**Pull request merged**